### ISSUE TITLE
Examples Requests with web proxy support

### ISSUE THREAD

**Issue by walterk82** [f5devcentral/vscode-f5#88]
I'm having difficulty getting the examples from github to load in VS.Code.  I have tried the URL in my browser and it loads fine.  Is there something missing that would cause web proxy servers to be bypassed?  Extension downloads thru VS.Code work just fine.

**Comment by DumpySquare** [comment]
Yes, you are correct.  The extension does not have any knowledge of proxies. and attempts to go direct.  

We can look into trying to detect and use system proxy settings, if not, look into provide a configuration option within the extension.

**Comment by DumpySquare** [comment]
@walterk82 I've been working on this as part of the v3.0 release for you.  Can you provide any details or example of your proxy configuration?  I know we need a destination:port, but I guess my main question is, do you also need to provide a username/password?

**Comment by walterk82** [comment]
Our proxy uses integrated auth (Kerberos) on Windows.  I would have thought VS.Code provide a framework for this type of communication.

**Comment by DumpySquare** [comment]
Right.  VSCode itself does have a way to tap into the local system network settings for proxy configuration but they do not provide a way for extensions to do the same, or make external HTTP calls through the core VSCode process (to utilize the local network/proxy settings).  

I will keep digging to see what I can come up with.  If we have to hold user creds for proxy auth, we can secure the password storage the same way we are storing passwords for connecting to F5 devices. Not ideal, but a workable solution.

**Comment by walterk82** [comment]
This seems overly complicated.  I think that using links to github and opening a browser version is just fine.  It keeps the security clean, simple, and less code for you to maintain.  I don't think you should have to re-write interacting with github in Node.JS for the conveniences of presenting these examples in a tree view.  I would hope that the github organization would be so straight forward to find the required examples.

**Comment by DumpySquare** [comment]
ok.  sounds like we can table this for now.  I still think there is value in having proxy support for enterprise environments like yours.  Does the ATC ILX RPM download/install function work for you (it downloads rpms from github)? 

https://f5devcentral.github.io/vscode-f5/#/atc_rpm_mgmt

**Comment by walterk82** [comment]
I don't have a quick way to test that as the RPMs are large and I don't load them over a VPN connection.  I also don't have a DEV VS.Code environment within the DataCenter to test with.  I have been loading them via GUI from a JumpBox.

**Comment by DumpySquare** [comment]
Ok.  I'm gonna push this feature for a later release.

**Comment by DumpySquare** [comment]
Hey @walterk82, are you still having this issue?  I'm hoping this might have gotten better with all the latest updates.

**Comment by walterk82** [comment]
Browsing the examples tree I see the following console output:

[2021-05-27T14:34:43.592Z] ERROR: {
  message: 'EXTERNAL-HTTPS-REQUEST-FAILED',
  path: undefined,
  err: 'Socket is closed'
}

**Comment by DumpySquare** [comment]
hrmmm.  ok.  thanks for the update
